Lib/idlelib/stackviewer.py

```diff
@@ -1,33 +1,31 @@
 import linecache
 import os
-import sys
 
 import tkinter as tk
 
 from idlelib.debugobj import ObjectTreeItem, make_objecttreeitem
 from idlelib.tree import TreeNode, TreeItem, ScrolledCanvas
 
-def StackBrowser(root, flist=None, tb=None, top=None):
+def StackBrowser(root, flist=None, exc=None, top=None):
     global sc, item, node  # For testing.
     if top is None:
         top = tk.Toplevel(root)
     sc = ScrolledCanvas(top, bg="white", highlightthickness=0)
     sc.frame.pack(expand=1, fill="both")
-    item = StackTreeItem(flist, tb)
+    item = StackTreeItem(flist, exc)
     node = TreeNode(sc.canvas, None, item)
     node.expand()
 
 
 class StackTreeItem(TreeItem):
 
-    def __init__(self, flist=None, tb=None):
+    def __init__(self, flist=None, exc=None):
+        assert isinstance(exc, BaseException)
         self.flist = flist
-        self.stack = self.get_stack(tb)
-        self.text = self.get_exception()
+        self.stack = self.get_stack(exc.__traceback__)
+        self.text = self.get_exception(exc)
```

Comment:
```suggestion
        self.text =  f"{type(exc).__name__}: {str(exc)}"
```
and delete self.get_exception, which is only called here.

Reply:
fixed and pushed June 7